### PR TITLE
[core] Added null check for output buffer

### PR DESCRIPTION
**Pull request by gaurav1086**
Check output != nullptr before calling output->clear()

### CODE REVIEW

File: tensorflow/core/lib/jpeg/jpeg_mem.cc [hunk 1]
```diff
@@ -593,6 +593,11 @@ bool GetImageInfo(const void* srcdata, int datasize, int* width, int* height,
 namespace {
 bool CompressInternal(const uint8* srcdata, int width, int height,
                       const CompressFlags& flags, tstring* output) {
+  if (output == nullptr)
+    LOG(ERROR) << "Output buffer is null: ";
+    return false;
```
Comment (Leslie-Fang) [inline tensorflow/core/lib/jpeg/jpeg_mem.cc]: Hi @gaurav1086 If we have find the NULL point, will should we just return false? How about create a new tstring object?

**Review by Leslie-Fang — CHANGES_REQUESTED** [review]
if we just barely return false may induce other following logic issue, right?

File: tensorflow/core/lib/jpeg/jpeg_mem.cc (hunk 1 above)
Comment (gaurav1086) [reply tensorflow/core/lib/jpeg/jpeg_mem.cc]: @Leslie-Fang , thanks a lot for the review. I appreciate your pointer, however I am under the impression (have to double check) that the caller of this function (GetImageInfo) should/would check the return value (bool: true/false), before trying to access the output buffer. Thoughts ? This function returns false at other occasions, hence I thought of returning  a false here as well.

**Comment by gaurav1086** [comment]
> if we just barely return false may induce other following logic issue, right?

@Leslie-Fang , thanks a lot for the review. The function CompressInternal() returns false in most of the error cases, so I believe the caller should handle it the similar way that it handles other errors. Thoughts ?

**Comment by gaurav1086** [comment]
Hi, is there anything from my side here ? Please advise. Thanks.

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@gaurav1086 Can you please fix the build failures? Thanks!

**Comment by gaurav1086** [comment]
@mihaimaruseac @gbaned , rebased with the latest master

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@gaurav1086 Can you please fix the build failures? Thanks!

**Comment by gaurav1086** [comment]
@gbaned fixed the build errors. Thank you.

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**